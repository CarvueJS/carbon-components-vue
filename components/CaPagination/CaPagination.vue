<template>
  <div class="bx--pagination" data-pagination>
    <!-- <div class="bx--pagination__left">
      <span class="bx--pagination__text">Items per page:</span>
      <div class="bx--select bx--select--inline">
        <label for="select-id-pagination" class="bx--visually-hidden">Number of items per page</label>
        <select id="select-id-pagination" class="bx--select-input" data-items-per-page>
          <option class="bx--select-option" value="20" v-for="n in numberOfPage" :key="n">
            {{n}}
          </option>
        </select>
        <svg class="bx--select__arrow" width="10" height="5" viewBox="0 0 10 5" fill-rule="evenodd">
          <path d="M10 0L5 5 0 0z"></path>
        </svg>
      </div>
      <span class="bx--pagination__text">
        <span>|&nbsp;</span>
        <span data-displayed-item-range>1-10</span> of
        <span data-total-items>40</span> items
      </span>
    </div> -->
    <div class="bx--pagination__right bx--pagination--inline">
      <span class="bx--pagination__text">
        <span data-displayed-page-number>{{value}}</span> of
        <span data-total-pages>{{numberOfPage}}</span> pages
      </span>
      <button class="bx--pagination__button bx--pagination__button--backward" data-page-backward aria-label="Backward button">
        <svg class="bx--pagination__button-icon" width="8" height="12" viewBox="0 0 8 12" fill-rule="evenodd">
          <path d="M7.5 10.6L2.8 6l4.7-4.6L6.1 0 0 6l6.1 6z"></path>
        </svg>
      </button>
      <label for="page-number-input" class="bx--visually-hidden">Page number input</label>
      <div class="bx--select bx--select--inline">
        <label for="select-id-pagination" class="bx--visually-hidden">Number of items per page</label>
        <select id="select-id-pagination" class="bx--select-input" data-page-number-input :value="value" @change="select">
          <option class="bx--select-option" v-for="n in numberOfPage" :key="n" >
            <div>{{n}}</div>
          </option>
        </select>
        <svg class="bx--select__arrow" width="10" height="5" viewBox="0 0 10 5" fill-rule="evenodd">
          <path d="M10 0L5 5 0 0z"></path>
        </svg>
      </div>
      <button class="bx--pagination__button bx--pagination__button--forward" data-page-forward aria-label="Forward button">
        <svg class="bx--pagination__button-icon" width="8" height="12" viewBox="0 0 8 12" fill-rule="evenodd">
          <path d="M0 10.6L4.7 6 0 1.4 1.4 0l6.1 6-6.1 6z"></path>
        </svg>
      </button>
    </div>
  </div>
</template>

<script>
import { Pagination } from 'carbon-components';

export default {
  name: 'ca-pagination',
  props: {
    value: {
      type: Number,
      default: 1,
    },
    prePage: {
      type: Number,
      default: 5,
    },
    numberOfItems: {
      type: Number,
      default: 100,
    },
  },
  data: () => ({
    pagination: null,
  }),
  computed: {
    numberOfPage() {
      return this.numberOfItems / this.prePage;
    },
  },
  mounted() {
    this.pagination = Pagination.create(this.$el);
    this.pagination.element.addEventListener('pageChange', (e) => {
      if (e.detail.direction === 'forward') {
        this.value > 1 ? this.$emit('input', this.value - 1) : '';
      } else {
        this.$emit('input', this.value + 1);
      }
    });
  },
  methods: {
    select(e) {
      this.$emit('input', parseInt(e.target.value));
    },
  },
};

</script>

<style></style>
