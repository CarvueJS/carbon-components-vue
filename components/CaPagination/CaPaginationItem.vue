<template>
  <option class="bx--select-option" value="20" :selected="selected"><slot></slot></option>
</template>

<script>
export default {
  name: 'ca-pagination-item',
  props: {
    selected: {
      type: Boolean,
      default: false,
    },
  },
};
</script>
