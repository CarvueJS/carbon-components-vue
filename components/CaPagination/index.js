import CaPagination from './CaPagination.vue';
import CaPaginationItem from './CaPaginationItem.vue';

CaPagination.install = (Vue) => {
  Vue.component(CaPagination.name, CaPagination);
  Vue.component(CaPaginationItem.name, CaPaginationItem);
};

export default CaPagination;
