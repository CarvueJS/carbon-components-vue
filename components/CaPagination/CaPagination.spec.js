import { shallow } from 'vue-test-utils';
import CaPagination from './CaPagination.vue';
import CaPaginationItem from './CaPaginationItem.vue';

describe('CaPagination', () => {
  test('is a Vue instance', () => {
    const wrapper = shallow(CaPagination, { 
      slots: {
        default: CaPaginationItem
      }
    });
    expect(wrapper.isVueInstance()).toBeTruthy();
  });
})
