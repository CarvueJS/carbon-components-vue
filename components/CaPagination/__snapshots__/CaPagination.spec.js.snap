// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaPagination Mathes snapshot 1`] = `
<div data-pagination="" class="bx--pagination">
  <div class="bx--pagination__right bx--pagination--inline"><span class="bx--pagination__text"><span data-displayed-page-number="">1</span> of
    <span data-total-pages="">20</span> pages
    </span>
    <button data-page-backward="" aria-label="Backward button" class="bx--pagination__button bx--pagination__button--backward">
      <svg width="8" height="12" viewBox="0 0 8 12" fill-rule="evenodd" class="bx--pagination__button-icon">
        <path d="M7.5 10.6L2.8 6l4.7-4.6L6.1 0 0 6l6.1 6z"></path>
      </svg>
    </button>
    <label for="page-number-input" class="bx--visually-hidden">Page number input</label>
    <div class="bx--select bx--select--inline">
      <label for="select-id-pagination" class="bx--visually-hidden">Number of items per page</label>
      <select id="select-id-pagination" data-page-number-input="" class="bx--select-input">
        <option class="bx--select-option">
          <div>1</div>
        </option>
        <option class="bx--select-option">
          <div>2</div>
        </option>
        <option class="bx--select-option">
          <div>3</div>
        </option>
        <option class="bx--select-option">
          <div>4</div>
        </option>
        <option class="bx--select-option">
          <div>5</div>
        </option>
        <option class="bx--select-option">
          <div>6</div>
        </option>
        <option class="bx--select-option">
          <div>7</div>
        </option>
        <option class="bx--select-option">
          <div>8</div>
        </option>
        <option class="bx--select-option">
          <div>9</div>
        </option>
        <option class="bx--select-option">
          <div>10</div>
        </option>
        <option class="bx--select-option">
          <div>11</div>
        </option>
        <option class="bx--select-option">
          <div>12</div>
        </option>
        <option class="bx--select-option">
          <div>13</div>
        </option>
        <option class="bx--select-option">
          <div>14</div>
        </option>
        <option class="bx--select-option">
          <div>15</div>
        </option>
        <option class="bx--select-option">
          <div>16</div>
        </option>
        <option class="bx--select-option">
          <div>17</div>
        </option>
        <option class="bx--select-option">
          <div>18</div>
        </option>
        <option class="bx--select-option">
          <div>19</div>
        </option>
        <option class="bx--select-option">
          <div>20</div>
        </option>
      </select>
      <svg width="10" height="5" viewBox="0 0 10 5" fill-rule="evenodd" class="bx--select__arrow">
        <path d="M10 0L5 5 0 0z"></path>
      </svg>
    </div>
    <button data-page-forward="" aria-label="Forward button" class="bx--pagination__button bx--pagination__button--forward">
      <svg width="8" height="12" viewBox="0 0 8 12" fill-rule="evenodd" class="bx--pagination__button-icon">
        <path d="M0 10.6L4.7 6 0 1.4 1.4 0l6.1 6-6.1 6z"></path>
      </svg>
    </button>
  </div>
</div>
`;
